### PR TITLE
Support batch operation with list of workflow executions

### PR DESCRIPTION
**Pull request by yux0**
**What changed?**
Support batch operation with list of workflow executions

**Why?**
Support batch operation with list of workflow executions

**How did you test it?**
Add unit tests and verify the function locally.

**Potential risks**

**Is hotfix candidate?**

### CODE REVIEW

**Review by MichaelSnowden — APPROVED** [review]
The changes LGTM, but I think we should add some test coverage before landing this

File: service/frontend/errors.go [hunk 1]
```diff
@@ -98,5 +98,8 @@ var (
 	errListNotAllowed      = serviceerror.NewPermissionDenied("List is disabled on this namespace.", "")
 	errSchedulesNotAllowed = serviceerror.NewPermissionDenied("Schedules are disabled on this namespace.", "")
 
-	errBatchAPINotAllowed = serviceerror.NewPermissionDenied("Batch operation feature are disabled on this namespace.", "")
+	errBatchAPINotAllowed                = serviceerror.NewPermissionDenied("Batch operation feature are disabled on this namespace.", "")
+	errBatchOpsWorkflowFilterNotSet      = serviceerror.NewInvalidArgument("Workflow executions and visibility filter are not set on request.")
+	errBatchOpsWorkflowFiltersNotAllowed = serviceerror.NewInvalidArgument("Workflow executions and visibility filter are all set on request. Only one of them is allowed.")
```
Comment (yycptt) [inline service/frontend/errors.go:103]: nit: all -> both

File: service/frontend/errors.go [hunk 2]
```diff
@@ -98,5 +98,8 @@ var (
...
+	errBatchOpsMaxWorkflowExecution      = serviceerror.NewInvalidArgument("Workflow executions size exceeded.")
```
Comment (yycptt) [inline service/frontend/errors.go:104]: nit: size -> count?

File: service/worker/batcher/workflow.go [hunk 3]
```diff
@@ -98,9 +98,11 @@ type (
 		Namespace string
 		// To get the target workflows for processing
 		Query string
+		// Target workflows for processing
+		Executions []*commonpb.WorkflowExecution
 		// Reason for the operation
 		Reason string
-		// Supporting: signal,cancel,terminate
+		// Supporting: signal,cancel,terminate,delete
```
Comment (yycptt) [inline service/worker/batcher/workflow.go:105]: sometimes I hope there's a type for workflow refresh as well :).

**Review by yycptt — APPROVED** [review]

File: service/worker/batcher/workflow.go (hunk 3 above)
Comment (yux0) [reply service/worker/batcher/workflow.go:105]: We should. Let's track this somewhere.

**Pull request merged**